pkg/scalers/kafka_scaler.go

```diff
@@ -336,17 +357,25 @@ func (s *kafkaScaler) getConsumerOffsets(topicPartitions map[string][]int32) (*s
 func (s *kafkaScaler) getLagForPartition(topic string, partitionID int32, offsets *sarama.OffsetFetchResponse, topicPartitionOffsets map[string]map[int32]int64) (int64, error) {
 	block := offsets.GetBlock(topic, partitionID)
 	if block == nil {
-		kafkaLog.Error(fmt.Errorf("error finding offset block for topic %s and partition %d", topic, partitionID), "")
-		return 0, fmt.Errorf("error finding offset block for topic %s and partition %d", topic, partitionID)
+		errMsg := fmt.Errorf("error finding offset block for topic %s and partition %d", topic, partitionID)
+		kafkaLog.Error(errMsg, "")
+		return 0, errMsg
 	}
 	consumerOffset := block.Offset
 	if consumerOffset == invalidOffset && s.metadata.offsetResetPolicy == latest {
-		kafkaLog.V(0).Info(fmt.Sprintf("invalid offset found for topic %s in group %s and partition %d, probably no offset is committed yet", topic, s.metadata.group, partitionID))
-		return invalidOffset, fmt.Errorf("invalid offset found for topic %s in group %s and partition %d, probably no offset is committed yet", topic, s.metadata.group, partitionID)
+		retVal := int64(1)
+		if s.metadata.scaleToZeroOnInvalidOffset {
+			retVal = 0
+		}
+		errMsg := fmt.Errorf(
+			"invalid offset found for topic %s in group %s and partition %d, probably no offset is committed yet. Returning with lag of %d",
+			topic, s.metadata.group, partitionID, retVal)
+		kafkaLog.V(0).Info(errMsg.Error())
```

Comment:
I wonder whether we want to return error?

Reply:
@zroubalik I have some topics that are inactive for some time and the offset is deleted (after a week ) therefore, when we receive another event to that topic keda doesn't work because it return -1 instead of 1. In my opinion, it should return 0 instead -1 (don't fail) because the offsets were deleted in the topic so it's ok to trace that if you want but it should return error, right?

Reply:
I might be wrong, but if you set the new property that's introduced in this PR, it should return 0 and for this case I am asking if we should print error. I might be missing something of course.

Reply:
I agree. Changed to not return error. This btw does not change the logic of the two places using this method as they either ignore the error or check that also the returning lag is -1

Reply:
Agree. Yeah, that's what I noticed as well.